source/Ship.cpp

```diff
@@ -3760,6 +3760,230 @@ void Ship::AddOutfit(const Outfit *outfit, int count)
+{
+	if(!targetAttribute.empty())
+	{
+		double limiter = 0.;
+		double minAttributeValue = 0.;
+		double originalBaseValue = baseAttributes.Get(targetAttribute);
+		double originalValue = attributes.Get(targetAttribute);
+		double newBaseValue = originalBaseValue + modifyAmount;
+		double newValue = originalValue + modifyAmount;
+
+		// Safety checks to ensure the new value is within parameters.
+		if(newBaseValue < 1. && newValue < 1. && targetAttribute == "hull")
+		{
+			// This means the minimum value for this attribute is 1.0
+			minAttributeValue = 1.;
+			limiter = minAttributeValue - newBaseValue;
+		}
+		else if(newBaseValue < 0.01 && (targetAttribute == "drag" || targetAttribute == "mass"))
+		{
+			// This means the minimum value for these attributes is 0.01
+			minAttributeValue = 0.01;
+			limiter = minAttributeValue - newBaseValue;
+		}
+		// Special handling for attributes that cannot be less than 0.
+		// These values can be 0, just they cannot be negative.
+		else if(newBaseValue < 0. && (targetAttribute == "outfit space" || targetAttribute == "cargo space" ||
+			targetAttribute == "weapon capacity" || targetAttribute == "engine capacity" ||
+			targetAttribute == "engine mod space" || targetAttribute == "shields" ||
+			targetAttribute == "reverse thruster slot" || targetAttribute == "steering slot" ||
+			targetAttribute == "thruster slot" || targetAttribute == "lateral thruster slot" ||
+			targetAttribute == "bunks" || targetAttribute == "fuel capacity" || targetAttribute == "required crew"))
+		{
+			if(newBaseValue < 0.)
+			{
+				minAttributeValue = 0.;
+				limiter = minAttributeValue - newBaseValue;
+			}
+		}
+
+		// Calculations take place here
+		newBaseValue += limiter;
+		newValue += limiter;
+
+		if(targetAttribute == "mass")
+		{
+			// Call the special method just for mass.
+			baseAttributes.ModifyMass(newBaseValue);
+			attributes.ModifyMass(newValue);
+		}
+		else
+		{
+			baseAttributes.Set(targetAttribute.c_str(), newBaseValue);
+			attributes.Set(targetAttribute.c_str(), newValue);
+		}
+
+		// Ensuring the current hull value is changed as well.
+		if(targetAttribute == "hull")
+		{
+			// Adds the modifyAmount and the limiter to the current hull too.
+			hull += modifyAmount + limiter;
+		}
+
+		// Ensuring the current shields value is changed as well.
+		if(targetAttribute == "shields")
+		{
+			// Adds the modifyAmount and the limiter to the current shields too.
+			shields += modifyAmount + limiter;
+		}
+
+		if(targetAttribute == "cargo space")
+		{
+			cargo.SetSize(attributes.Get("cargo space"));
+
+			// Only the player's ships make use of attraction and deterrence.
+			if(isYours)
+				attraction = CalculateAttraction();
+		}
+
+		// If the added or removed attribute is hyperdrive, scram drive, or jump drive capability, then
+		// recalibrate the navigation.
+		if((targetAttribute == "hyperdrive" || targetAttribute == "scram drive" || targetAttribute == "jump drive"))
```

Comment:
What about fuel? I forgot how this works tbh

Reply:
I could add fuel too, I suppose, but the main purpose of this little code snipped is to catch instances of where the means of doing a jump changed and now there's a different drive that is the one that should be used.

Reply:
Oh yeah true dont matter if the color for amount of jumps you can make is wrong a bit
I'd test to make sure it fixes itself eventually but it probably will? Idk